scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -106,20 +106,25 @@ class RunCoverage(
   fun execute() {
     val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptionTextProto)
       .testFileExemptionList
-      .filter { it.testFileNotRequired }
+      .filter { it.testFileNotRequired || it.sourceFileIsIncompatibleWithCodeCoverage }
```

Comment:
@Rd4dev can you please advise on how I can best test this? If there's a specific test that I can copy in order to verify that source files are correctly filtered when this exemption is enabled, that would be really helpful.

Reply:
Until PR 2.1 (which this PR is tested on), I used the available `test_file_exemptions.textproto` data to test exempted files, it was with PR 2.2 I introduced providing **custom test exemption list** as a parameter to test them (as to verify overridden percentages).

I think may be having a custom list would be a way to test in this scenerio.

I added the implementation and the test case for exempting incompatible source files: [**[implementation code]**](https://github.com/Rd4dev/oppia-android/pull/7/commits/8c9cdd44b6b1b7f2fc7e67115c39fa6e72634aa4#diff-5a779ea4a8aa2caf5249ced27465f76b8ac19bed5cf250475f6033dfad464908R222), can you please take a look at the above test case and see if that could be appropriate?